### PR TITLE
[model] fix: dit + lora related fix

### PR DESCRIPTION
**Pull request by Coach257**
## Dit trainer related:
1. support sp offline embedding
2. fix helper when token_len == 0 (offline embedding has no token)
## Lora related: (only dit tested currently)
1. fix fsdp1 meta/gpu init load lora_adapter & init lora
2. add fsdp2 meta init load peftmodel & load adapter & init lora
3. del `training architecture`, only keep `model.lora_config`
4. dcp shard save lora
## Others:
1. fix optimizer None state when some modules have no grad
2. add save utils for video with audios

### CODE REVIEW

File: veomni/utils/lora_utils.py [hunk 1]
```diff
@@ -97,3 +101,137 @@ def load_state_dict_from_bin(file_path, torch_dtype=None):
...
+        for piece in pieces:
+            if piece.startswith("lora_"):
+                break
+            lora_layer_fqn_parts.append(piece)
+            lora_layer = getattr(lora_layer, piece)
+        lora_layer_fqn = ".".join(lora_layer_fqn_parts)
+        if lora_layer_fqn not in _initialized and hasattr(lora_layer, "reset_lora_parameters"):
+            _initialized.add(lora_layer_fqn)
+            for adapter in getattr(lora_layer, "lora_A", {}).keys():
+                lora_layer.reset_lora_parameters(adapter, init_lora_weights=True)
```
Comment (FoolPlayer) [inline veomni/utils/lora_utils.py:160]: Init lora per rank will use the same random seed, every rank will get the same initial weight. It's better just init on rank0 and shard to other ranks.

File: veomni/trainer/callbacks/checkpoint_callback.py [hunk 2]
```diff
@@ -179,15 +181,14 @@ def _save_checkpoint(self, state: TrainerState, stage: str = "step_end"):
             self.trainer.checkpointer.save_future.result()
 
         if stage == "train_end":
-            del self.trainer.optimizer
-            del self.trainer.lr_scheduler
+            self.trainer.optimizer = None
```
Comment (FoolPlayer) [inline veomni/trainer/callbacks/checkpoint_callback.py:184]: Why change to `= None`

File: tasks/deprecated_task/train_flux.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 )
 
 from veomni.arguments import DataArguments, ModelArguments, TrainingArguments, parse_args, save_args
-from veomni.checkpoint import build_checkpointer
+from veomni.checkpoint import build_checkpointer, ckpt_to_state_dict
```
Comment (FoolPlayer) [inline tasks/deprecated_task/train_flux.py:17]: Do we still use the deprecated tasks?

File: veomni/trainer/callbacks/checkpoint_callback.py (hunk 2 above)
Comment (Coach257) [reply veomni/trainer/callbacks/checkpoint_callback.py:184]: https://github.com/ByteDance-Seed/VeOmni/pull/735#discussion_r3199080363 Gemini comments here

File: tasks/deprecated_task/train_flux.py (hunk 3 above)
Comment (Coach257) [reply tasks/deprecated_task/train_flux.py:17]: Agent fixed these automatically 😀

File: veomni/utils/lora_utils.py (hunk 1 above)
Comment (Coach257) [reply veomni/utils/lora_utils.py:160]: DTensor random ops are shard-aware, nn.init_weight generates param based on (seed + shard_offset).

**Review by FoolPlayer — APPROVED** [review]
